rm/rm-client/src/main/java/org/ow2/proactive/resourcemanager/common/event/RMInitialState.java

```diff
@@ -163,41 +147,42 @@ private void updateCounter(RMEvent event) {
      * @param filter
      * @return rmInitialState where all the events bigger than 'filter'
      */
-    public RMInitialState cloneAndFilter(long filter) {
-        long actualFilter;
-        if (filter <= latestCounter.get()) {
-            actualFilter = filter;
+    public RMStateDTO cloneAndFilter(long filter) {
+        long actualFilter = computeActualFilter(filter);
+
+        final List<RMEvent> responseEvents = events.getSortedItems()
+                                                   .tailSet(new RMEvent(actualFilter + 1)) // because tailSet returns event which is equal or greater
```

Comment:
Instead of adding one here, we could make computeActualFilter return already the value that includes the first element of the tailSet.

Reply:
Forget about my comment above, I see that actualFilter is used after.

Reply:
I would not do this. Filter is exclusive, when tailSet is inclusive. I would keep it to make it obvious